core/src/main/java/org/infinispan/cache/impl/DecoratedCache.java

```diff
@@ -267,16 +299,22 @@ public boolean replace(K key, V oldValue, V value, long lifespan, TimeUnit lifes
             .maxIdle(maxIdle, maxIdleUnit)
             .build();
 
-      return cacheImplementation.putAsync(key, value, metadata, flags);
+      return putAsync(key, value, metadata);
    }
 
    @Override
    public CompletableFuture<Void> putAllAsync(Map<? extends K, ? extends V> data) {
+      if (lockOwner != null) {
```

Comment:
I had thought about that, but couldn't decide. I can make a private method where it passes the method name and do string concat in the exception then.